lib/assert.js

```diff
@@ -885,6 +886,34 @@ assert.ifError = function ifError(err) {
   }
 };
 
+assert.match = function match(string, regexp, message) {
+  if (!isRegExp(regexp)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'regexp', 'RegExp', regexp
+    );
+  }
+  if (typeof string !== 'string' || !RegExpPrototypeTest(regexp, string)) {
+    const generatedMessage = !message;
+    message = message || (typeof string !== 'string' ?
```

Comment:
Should we validate the type of `message` a bit?

Reply:
We do not do that anywhere else so far and I would prefer not to do that. I would actually like to allow the message to be overloaded at some point so that it provides additional information for the user.

I think we should at some point combine user messages and automatically generated ones. Similar to the way we handle it in `assert.throws`, etc.

In case it would not be a string but e.g., an object, we could then inspect that value and add something to the message along the lines of: `Additional information passed through: <inspected object>`.